cns/kubecontroller/nodenetworkconfig/conversion_windows.go

```diff
@@ -10,28 +10,26 @@ import (
 )
 
 // createNCRequestFromStaticNCHelper generates a CreateNetworkContainerRequest from a static NetworkContainer.
-// If the NC's DefaultGateway is empty, it will set the 0th IP as the gateway IP and all remaining IPs as
-// secondary IPs. If the gateway is not empty, it will not reserve the 0th IP and add it as a secondary IP.
+// If the NC's DefaultGateway is empty and nc type is overlay, it will set the 2nd IP (*.1) as the gateway IP and all remaining IPs as
+// secondary IPs. If the gateway is not empty, it will not reserve the 2nd IP and add it as a secondary IP.
 //
 //nolint:gocritic //ignore hugeparam
 func createNCRequestFromStaticNCHelper(nc v1alpha.NetworkContainer, primaryIPPrefix netip.Prefix, subnet cns.IPSubnet) (*cns.CreateNetworkContainerRequest, error) {
 	secondaryIPConfigs := map[string]cns.SecondaryIPConfig{}
-
-	// if NC DefaultGateway is empty, set the 0th IP to the gateway and add the rest of the IPs
-	// as secondary IPs
-	startingAddr := primaryIPPrefix.Masked().Addr() // the masked address is the 0th IP in the subnet
+	// the masked address is the 0th IP in the subnet and startingAddr is the 2nd IP (*.1)
+	startingAddr := primaryIPPrefix.Masked().Addr().Next()
+	// if NC DefaultGateway is empty, set the 2nd IP (*.1) to the gateway and add the rest of the IPs as secondary IPs
 	if nc.DefaultGateway == "" && nc.Type == v1alpha.Overlay {
-		// assign 0th IP to the default gateway
 		nc.DefaultGateway = startingAddr.String()
 		startingAddr = startingAddr.Next()
-	} else if nc.Type == v1alpha.VNETBlock {
-		// skipping 0th IP for the Primary IP of NC
-		startingAddr = startingAddr.Next()
 	}
 
 	// iterate through all IP addresses in the subnet described by primaryPrefix and
 	// add them to the request as secondary IPConfigs.
 	for addr := startingAddr; primaryIPPrefix.Contains(addr); addr = addr.Next() {
+		if addr.Is4() && addr.As4()[3] == 255 {
```

Comment:
it's not really the .255 address that is reserved, it's the last address in the subnet. This will work I think because the smallest subnet that can be used for overlay is /24 and every CIDR large than that's last address ends in 255, but I'd suggest computing the last address and skipping that instead of skipping the "255th" address, in case we ever decide to use smaller pod CIDRs for overlay

Reply:
HNS told us to skip all .255s but yeah we can just skip last one here if the subnet will never get larger

Reply:
let me confirm with them again on 'last' vs .255 of every subnet

Reply:
ok updated to skip last IP of subnet